app/controllers/users/two_factor_verification_controller.rb

```diff
@@ -0,0 +1,57 @@
+module Users
+  class TwoFactorVerificationController < ApplicationController
+    before_action :check_direct_otp_expiry, only: [:show, :create]
+
+
+    def show
+      unless current_user.totp_enabled?
+        current_user.create_direct_otp
+        r = SMSService.new.send_two_factor_confirmation_code!(current_user)
+
+        # TODO: make this async and handle service failure
+        unless r.code == 202
+          raise 'There was an error sending the SMS'
+        end
+      end
+    end
+
+    def create
```

Comment:
Really good point. I guess it'd be stopped at the outer by needing to be signed in, but your point still stands.